pkg/scalers/azure_blob_scaler.go

```diff
@@ -142,14 +142,7 @@ func parseAzureBlobMetadata(config *ScalerConfig, logger logr.Logger) (*azure.Bl
 	if val, ok := config.TriggerMetadata["useAAdPodIdentity"]; ok && config.PodIdentity.Provider == "" && val == stringTrue {
 		config.PodIdentity = kedav1alpha1.AuthPodIdentity{Provider: kedav1alpha1.PodIdentityProviderAzure}
 	}
-
-	// FIXME: DEPRECATED to be removed in v2.12
-	if val, ok := config.TriggerMetadata["metricName"]; ok {
-		meta.MetricName = kedautil.NormalizeString(fmt.Sprintf("azure-blob-%s", val))
-	} else {
-		meta.MetricName = kedautil.NormalizeString(fmt.Sprintf("azure-blob-%s", meta.BlobContainerName))
-	}
-
+	meta.MetricName = kedautil.NormalizeString(fmt.Sprintf("azure-blob-%s", meta.BlobContainerName))
```

Comment:
Do we need to generate this here? I mean, if we have only one option for `MetricName` and it's used once, maybe we could remove the object variable and generate the name wherever is used. WDYT? (If you agree, this also applies to other scalers)

Reply:
Yes. I just did another round of refactoring to remove this `metricName` object var. Added a few test cases. Please help me review it :pray: